File: py/stop.py
from argparse import ArgumentParser
import requests
import ujson as json


def parse_args():
    ap = ArgumentParser()
    ap.add_argument('--server_host', type=str, default='localhost',
                    help='Server host.')
    ap.add_argument('--server_port', type=int, default=1337,    
                    help='Server port.')
    return ap.parse_args()


def run(args):
    url = 'http://%s:%d/stop' % (args.server_host, args.server_port)
    ret = requests.post(url)
    print('Status code: %d' % ret.status_code)
    print('Body: %s' % ret.text)


if __name__ == '__main__':
    run(parse_args())

File: src/stop.cc
#include <cstdio>
#include <gflags/gflags.h>

#include "ext/cpr.h"
#include "ext/json.h"
#include "ext/sprintf.h"

DEFINE_string(server_host, "localhost", "Server host");
DEFINE_int32(server_port, 1337, "Server port");

int main(int argc, char* argv[]) {
    gflags::ParseCommandLineFlags(&argc, &argv, true);
    auto to = StringPrintf("http://%s:%d/stop", FLAGS_server_host.c_str(),
                           FLAGS_server_port);
    auto ret = cpr::Post(cpr::Url(to), cpr::Body{},
                         cpr::Header{{"Content-Type", "text/plain"}});
    printf("Status code: %d\n", ret.status_code);
    printf("Body: %s\n", ret.text.c_str());
}
